code/src/test/kotlin/com/expediagroup/sdk/core/http/ResponseTest.kt

```diff
@@ -0,0 +1,343 @@
+    fun `should set one single header`() {
+        // Given
+        val request = Request.Builder().method("GET").url("https://example.com").build()
+        val headerName = "content-type"
+        val headerValue = "application/json"
+
+        // When
+        val response = Response
+            .builder()
+            .request(request)
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.OK)
+            .message("OK")
+            .setHeader(headerName, headerValue)
+            .build()
+
+        // Expect
+        assertEquals(headerValue, response.headers.get(headerName))
+    }
+
+    @Test
+    fun `should set multiple values for one header`() {
+        // Given
+        val request = Request.Builder().method("GET").url("https://example.com").build()
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val response = Response
+            .builder()
+            .request(request)
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.OK)
+            .message("OK")
+            .setHeader(headerName, headerValue1)
+            .setHeader(headerName, headerValue2)
+            .build()
+
+        // Expect
+        assertEquals(listOf(headerValue2), response.headers.values(headerName))
+    }
+
+    @Test
+    fun `should add multiple values for one header as a list`() {
+        // Given
+        val request = Request.Builder().method("GET").url("https://example.com").build()
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val response = Response
+            .builder()
+            .request(request)
+            .protocol(Protocol.HTTP_1_1)
+            .status(Status.OK)
+            .message("OK")
+            .addHeader(headerName, listOf(headerValue1, headerValue2))
+            .build()
+
+        // Expect
+        assertEquals(listOf(headerValue1, headerValue2), response.headers.values(headerName))
+    }
+
+    @Test
+    fun `should set multiple values for one header as a list`() {
+        // Given
+        val request = Request.Builder().method("GET").url("https://example.com").build()
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val response = Response
+            .builder()
+            .request(request)
+            .protocol(Protocol.HTTP_1_1)
+            .message("OK")
+            .status(Status.OK)
+            .setHeader(headerName, listOf(headerValue1, headerValue2))
+            .setHeader(headerName, listOf(headerValue2, headerValue1))
+            .build()
+
+        // When
+        assertEquals(listOf(headerValue2, headerValue1), response.headers.values(headerName))
+    }
+
+    @Test
+    fun `should handle adding and setting headers as expected`() {
```

Comment:
Would it be feasible if we replaced `as expected` with the actual desired behavior?

Reply:
Removed the whole test, it's not clear with the expected behaviour and covered by other smaller tests